### PR TITLE
Simplify HDF5 and Mojo datasources for single channel

### PR DESCRIPTION
**Pull request by thejohnhoffer**
All datasources should only represent one channel. Currently Hdf5 and Mojo remain as the only datasources that allow the possibility of multiple image channels within them. 

This update removes a lot of the complexity previously added to support different channels in the hdf5 and mojo datasources.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -53,6 +53,10 @@ coverage.xml
 # Sphinx documentation
 docs/_build/
 
+# Remote debugging
+butterfly/debug.py
```
Comment (thejohnhoffer) [inline .gitignore:57]: I use ipdb and vim over an ssh connection. These are needed for those purposes.

File: butterfly/hdf5.py [hunk 2]
```diff
@@ -19,77 +19,56 @@
...
+
     The dataset may be sparse (with zero for areas w/o data) so the coordinate
     system is implicit, starting at 0, 0, 0.
     '''
-    
+
     def __init__(self, core, datapath, dtype=np.uint8):
-        layers = self.defaultLayers(datapath)
-        if not layers:
+        self._dataset = self.loadFolder(datapath)
```
Comment (thejohnhoffer) [inline butterfly/hdf5.py:33]: There are no layers now. There is only one self._dataset.

File: butterfly/hdf5.py [hunk 3]
```diff
@@ -19,77 +19,56 @@
...
-        return K_PRESET_CHANNELS[int(id)%len(K_PRESET_CHANNELS)]
-
-    def storeChannels(self,layers):
-        channels = {}
-        for lid,layer in enumerate(layers):
-            if K_CHANNEL not in layer:
-                layer[K_CHANNEL] = self.kindGuess(lid)
-            channels[layer[K_CHANNEL]] = layer
-        return channels
+            return json.load(open(path, "r"))
```
Comment (thejohnhoffer) [inline butterfly/hdf5.py:41]: Allow a .json file specifying the filename and the internal path

File: butterfly/hdf5.py [hunk 4]
```diff
@@ -10,86 +10,61 @@
...
+        elif path.endswith('.h5'):
+            with h5py.File(path, "r") as fd:
+                key0 = fd.keys()[0]
+                return {
+                    K_FILENAME : path,
+                    K_DATASET_PATH : key0,
+                }
+        else:
+            return False
```
Comment (thejohnhoffer) [inline butterfly/hdf5.py:50]: Allow an .h5 file by choosing the first internal path

**Comment by LeeKamentsky** [comment]
A big improvement that it's so much simpler... unfortunate that I will make it a bit more complicated now.

**Pull request merged**